File: databinding/src/main/java/com/example/anvil/databinding/TimePickerLayout.java
package trikita.anvil.listenertest;

import android.content.Context;
import android.widget.LinearLayout;

import static trikita.anvil.DSL.*;
import trikita.anvil.RenderableView;

public class TimePickerLayout extends RenderableView {

    int h = 0;
    int m = 0;

    public TimePickerLayout(Context c) {
        super(c);
    }

    @Override
    public void view() {
        linearLayout(() -> {
            size(FILL, FILL);
            orientation(LinearLayout.VERTICAL);

            timePickerLayout();
            timePickerLayout();
        });
    }

    private void timePickerLayout() {
        timePicker(() -> {
            size(WRAP, WRAP);
            weight(1);
            layoutGravity(CENTER);

            currentHour(h);
            currentMinute(m);

            onTimeChanged((picker, hr, min) -> {
                h = hr;
                m = min;
            });
        });
    }
}

File: databinding-kotlin/src/main/kotlin/com/example/anvil/databinding/TimePickerLayout.kt
package com.example.anvil.databinding

import android.content.Context
import android.widget.LinearLayout

import trikita.anvil.DSL.*
import trikita.anvil.RenderableView

class TimePickerLayout(c: Context) : RenderableView(c) {

    var h = 0
    var m = 0

    override fun view() {
        linearLayout {
            size(FILL, FILL)
            orientation(LinearLayout.VERTICAL)

            timePickerLayout()
            timePickerLayout()
        }
    }

    private fun timePickerLayout() {
        timePicker {
            size(WRAP, WRAP)
            weight(1f)
            layoutGravity(CENTER)

            currentHour(h)
            currentMinute(m)

            onTimeChanged { picker, hr, min ->
                h = hr
                m = min
            }
        }
    }
}

